### PR TITLE
keep env_status file and add fields in the report section to do auto redeploy

### PR DESCRIPTION
**Pull request by liyaqin1**
We may have trouble by deleting a service in file env_status. In this PR, on deployd side, we set RESET_BY_SYSTEM to trigger redeploy just like Teletraan UI did by clicking "Retry".

### CODE REVIEW

**Comment by ntascii** [comment]
we might need a different state to classify between Manual reset from user (RESET) and auto RESET by bad health.

File: deploy-agent/deployd/agent.py [hunk 1]
```diff
@@ -125,30 +125,37 @@ def serve_build(self):
             self._executor = Executor(callback=PingServer(self), config=self._config)
         # include healthStatus info for each container
         if len(self._envs) > 0:
-            need_to_delete = []
+            #need_to_delete = []
```
Comment (ntascii) [inline deploy-agent/deployd/agent.py:128]: remove commented code?

Comment (liyaqin1) [reply deploy-agent/deployd/agent.py:128]: done

**Comment by liyaqin1** [comment]
> we might need a different state to classify between Manual reset from user (RESET) and auto RESET by bad health.

sounds good, thanks

**Comment by liyaqin1** [comment]
Tested on my devapp and everything on deployd side looks correct.
When manually triggering a bad container, the redeploy times increased by 1 and the state is RESET_BY_SYSTEM:
Screenshot 2023-11-03 at 9 49 44 PM
On TAS side, since my devapp is not really having environments on dev1, it is not possible to trigger the redeployment.
But if I manually edit the state to be RESET_BY_SYSTEM for an existing stage on dev1, I can see it goes to the path as expected:
Screenshot 2023-11-03 at 9 55 45 PM

**Review by ntascii — APPROVED** [review]

**Pull request merged**